Feature/cmaq54 fengsha

bbakernoaa opened the pull request:
<!--Instructions: All subsequent sections of text should be filled in as appropriate.-->
## PR Checklist
- [ ] This PR has been tested on an RDHPCS machine and/or WCOSS2. Please select below:
  - [ ] RDHPCS.
  - [ ] WCOSS2.

- [ ] This PR has been tested with the ufs-srweather-app workflow online-cmaq branch.

- [ ] New or updated input data is required by this PR. <!-- If checked, please work with the code managers to update input data sets on all platforms.-->

- [ ] Baselines are expected to change.

## Description
<!--Provide a detailed description of what this PR does.-->

## Issue(s) addressed
<!--Please link the issues to be closed with this PR.
(Remember, issues must always be created before starting work on a PR branch!)
EX: - fixes #<issue_number>-->

## Dependencies
<!--If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).
Do PRs in upstream/other repositories need to be merged along with this one?
EX: - waiting on noaa-emc/fv3atm/pull/<pr_number>-->


this needs to be paired with the updated FV3 component as well as the vtype needs to be added to the chemistry export fields.  I have a branch here: https://github.com/bbakernoaa/fv3atm/tree/feature/vtype

bbakernoaa committed:
adding vtype

bbakernoaa committed:
add vegetation type

bbakernoaa committed:
remove fractional vegetation for now

bbakernoaa committed:
fix small error

bbakernoaa committed:
minor updates

drnimbusrain commented:
This pull request includes multiple changes across several modules to enhance the handling of new fields and improve the dust emission calculations. The most important changes include adding new parameters and variables, updating subroutines to handle these new variables, and implementing new functions for drag partitioning.

### Enhancements to field handling:

* [`src/aqm_cap.F90`](diffhunk://#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L16-R16): Increased `importFieldCount` to 37 and added `vtype` to `importFieldNames`. [[1]](diffhunk://#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L16-R16) [[2]](diffhunk://#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L55-R57)

### Updates to subroutines:

* [`src/aqm_comp_mod.F90`](diffhunk://#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cR607-R618): Updated `aqm_comp_import` subroutine to handle the new `vtype` and `vegetation_type_frac` fields. [[1]](diffhunk://#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cR607-R618) [[2]](diffhunk://#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cL626-R691)

### Dust emission calculations:

* [`src/model/src/DUST_EMIS.F`](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R117-R118): Added new variables `drag_opt` and `drylimit_factor`, and updated the `get_dust_emis` subroutine to include new logic for drag partitioning. [[1]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R117-R118) [[2]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L639-R642) [[3]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L753-R755) [[4]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R768) [[5]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R778-R789) [[6]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R814-R827) [[7]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L832-R857) [[8]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L1320) [[9]](diffhunk://#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R1429-R1590)

### Additional updates:

* [`src/model/src/ASX_DATA_MOD.F`](diffhunk://#diff-c5273473e62c85283711abdefae6a3eb9e7486f811f7a3e769b0f80f41ed9c75R201): Added `DLUSE` variable to store dominant land use information.
* [`src/shr/aqm_internal_mod.F90`](diffhunk://#diff-0342daa9c2c78a8dabc99835b49b0b2fdd8ac9410dbbfcaee3ad62a992bb200bR33): Added `dragopt` variable to the module.
* [`src/shr/aqm_methods.F90`](diffhunk://#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R486-R488): Updated the `ENVREAL` function to handle the new `CTM_WBDUST_FENGSHA_DRAGOPT` case and updated the `interpx` function to handle the new `DLUSE` case. [[1]](diffhunk://#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R486-R488) [[2]](diffhunk://#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15L694-R697) [[3]](diffhunk://#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R803-R804)


@bbakernoaa can we close this though?  Funny co-pilot review doesn't want to review fortran here...

bbakernoaa commented on src/model/src/DUST_EMIS.F:
```diff
@@ -796,10 +811,20 @@ subroutine get_dust_emis( jdate, jtime, tstep, rjacm, cellhgt,
      &        (Met_Data%rn( c,r ) + Met_Data%rc( c,r ) .le. 0.0254) .and. ! rn, rc = [cm]
      &        (Met_Data%snocov( c,r ) .lt. 0.001) .and.
-     &        (Met_Data%drag( c,r ) .gt. 0.0) ) then ! less than 0.1% snow coverage
+     &        (R .gt. 0.0) ) then ! less than 0.1% snow coverage
```
shouldn't this be `drag` and **NOT**  `R` as it is set just above this?

bbakernoaa commented:
we can close it for now.

Barry Baker

National Oceanic and Atmospheric Administration
Air Resources Laboratory
Physical Research Scientist
Chemical Modeling and Emissions Group Leader
NCWCP, R/ARL, Rm. 4204
5830 University Research Court
College Park, Maryland  20740
Phone: ‪(301) 683-1395‬



On Mon, Mar 31, 2025 at 12:02 PM Patrick Campbell ***@***.***>
wrote:

> This pull request includes multiple changes across several modules to
> enhance the handling of new fields and improve the dust emission
> calculations. The most important changes include adding new parameters and
> variables, updating subroutines to handle these new variables, and
> implementing new functions for drag partitioning.
> Enhancements to field handling:
>
>    - src/aqm_cap.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L16-R16>:
>    Increased importFieldCount to 37 and added vtype to importFieldNames.
>    [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L16-R16>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L55-R57>
>
> Updates to subroutines:
>
>    - src/aqm_comp_mod.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cR607-R618>:
>    Updated aqm_comp_import subroutine to handle the new vtype and
>    vegetation_type_frac fields. [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cR607-R618>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cL626-R691>
>
> Dust emission calculations:
>
>    - src/model/src/DUST_EMIS.F
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R117-R118>:
>    Added new variables drag_opt and drylimit_factor, and updated the
>    get_dust_emis subroutine to include new logic for drag partitioning.
>    [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R117-R118>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L639-R642>
>    [3]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L753-R755>
>    [4]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R768>
>    [5]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R778-R789>
>    [6]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R814-R827>
>    [7]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L832-R857>
>    [8]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L1320>
>    [9]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R1429-R1590>
>
> Additional updates:
>
>    - src/model/src/ASX_DATA_MOD.F
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-c5273473e62c85283711abdefae6a3eb9e7486f811f7a3e769b0f80f41ed9c75R201>:
>    Added DLUSE variable to store dominant land use information.
>    - src/shr/aqm_internal_mod.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-0342daa9c2c78a8dabc99835b49b0b2fdd8ac9410dbbfcaee3ad62a992bb200bR33>:
>    Added dragopt variable to the module.
>    - src/shr/aqm_methods.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R486-R488>:
>    Updated the ENVREAL function to handle the new
>    CTM_WBDUST_FENGSHA_DRAGOPT case and updated the interpx function to
>    handle the new DLUSE case. [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R486-R488>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15L694-R697>
>    [3]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R803-R804>
>
> @bbakernoaa <https://github.com/bbakernoaa> can we close this though?
> Funny co-pilot review doesn't want to review fortran here...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noaa-oar-arl/AQM/pull/25#issuecomment-2766687095>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFIUVNZSY54KIG3JSK2MMYD2XFRITAVCNFSM6AAAAABRJSZ5JKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRWGY4DOMBZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: drnimbusrain]*drnimbusrain* left a comment (noaa-oar-arl/AQM#25)
> <https://github.com/noaa-oar-arl/AQM/pull/25#issuecomment-2766687095>
>
> This pull request includes multiple changes across several modules to
> enhance the handling of new fields and improve the dust emission
> calculations. The most important changes include adding new parameters and
> variables, updating subroutines to handle these new variables, and
> implementing new functions for drag partitioning.
> Enhancements to field handling:
>
>    - src/aqm_cap.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L16-R16>:
>    Increased importFieldCount to 37 and added vtype to importFieldNames.
>    [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L16-R16>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-1b18b3a2fc2d6b2f584eb1ca949e71ec3bfb2a18764a1b97d40afeb37cc55fe4L55-R57>
>
> Updates to subroutines:
>
>    - src/aqm_comp_mod.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cR607-R618>:
>    Updated aqm_comp_import subroutine to handle the new vtype and
>    vegetation_type_frac fields. [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cR607-R618>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-cc22a1ec93d9fbd5ee5268d66cb983898abc3d3d1b9281a84c1bf075b47d1b2cL626-R691>
>
> Dust emission calculations:
>
>    - src/model/src/DUST_EMIS.F
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R117-R118>:
>    Added new variables drag_opt and drylimit_factor, and updated the
>    get_dust_emis subroutine to include new logic for drag partitioning.
>    [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R117-R118>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L639-R642>
>    [3]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L753-R755>
>    [4]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R768>
>    [5]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R778-R789>
>    [6]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R814-R827>
>    [7]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L832-R857>
>    [8]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7L1320>
>    [9]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-af0cd4716200930b73ebd6618388dc7856b329e900be7ee5128012e9cbe442e7R1429-R1590>
>
> Additional updates:
>
>    - src/model/src/ASX_DATA_MOD.F
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-c5273473e62c85283711abdefae6a3eb9e7486f811f7a3e769b0f80f41ed9c75R201>:
>    Added DLUSE variable to store dominant land use information.
>    - src/shr/aqm_internal_mod.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-0342daa9c2c78a8dabc99835b49b0b2fdd8ac9410dbbfcaee3ad62a992bb200bR33>:
>    Added dragopt variable to the module.
>    - src/shr/aqm_methods.F90
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R486-R488>:
>    Updated the ENVREAL function to handle the new
>    CTM_WBDUST_FENGSHA_DRAGOPT case and updated the interpx function to
>    handle the new DLUSE case. [1]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R486-R488>
>    [2]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15L694-R697>
>    [3]
>    <https://github.com/noaa-oar-arl/AQM/pull/25/files#diff-ae86eb8dcd46347a339a36a3753a9b28492431009c820d8624ddaf4080c17c15R803-R804>
>
> @bbakernoaa <https://github.com/bbakernoaa> can we close this though?
> Funny co-pilot review doesn't want to review fortran here...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/noaa-oar-arl/AQM/pull/25#issuecomment-2766687095>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFIUVNZSY54KIG3JSK2MMYD2XFRITAVCNFSM6AAAAABRJSZ5JKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRWGY4DOMBZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
